### PR TITLE
Remove demo code

### LINKED ISSUE #3531: Remove demo code
For the beta release, the demo logic and associated code will no longer be in the Mathesar service. It would be good to remove that code _before_ doing major work on permissions, etc. since it would be more difficult to maintain the demo while also overhauling permissions.

### PR DESCRIPTION
**Pull request by mathemancer**
Fixes #3531 

This PR removes all Demo code which is otherwise unused from Mathesar.

**Technical details**

Some demo code was used for example datasets in other context, and so was moved to a different location rather than removed entirely.

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `develop` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [X] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
I assume we want to wait until the April release branch is cut (cc @ghislaineguerin) before we merge this.

**Comment by mathemancer** [comment]
> I assume we want to wait until the April release branch is cut (cc @ghislaineguerin) before we merge this.

Definitely.

@pavish Any opinions on whether we should remove the demo-mode-related translations? I know we paid for them, but we can always recover them through git or other means when/if we want them back.

**Comment by pavish** [comment]
@mathemancer 

> Any opinions on whether we should remove the demo-mode-related translations? I know we paid for them, but we can always recover them through git or other means when/if we want them back.

Yes, we should remove any key we're no longer using.

**Comment by kgodey** [comment]
Agreed that paying for the translations isn't a reason to keep them around if we no longer need them.

**Comment by mathemancer** [comment]
Okay, this is now ready for review. From @pavish , I'm just requesting a quick check to make sure there aren't any obvious problems where the front end might break because it expects something (e.g., a flag or variable) to be available from the back end.

File: Dockerfile
Comment (pavish) [inline Dockerfile:120]: Since we no longer have the `demo` stage, I think we should remove the `common` stage as well.

File: mathesar/templates/mathesar/login_base.html
Comment (pavish) [inline mathesar/templates/mathesar/login_base.html:149]: We would have to figure out a different way to mention that the demo data is reset regularly to users accessing the demo site. Do we have an approach figured out on how to do that?

**Review by pavish — CHANGES_REQUESTED** [review]
@mathemancer The code changes look good to me. I have a couple comments.

File: mathesar/templates/mathesar/login_base.html
Comment (mathemancer) [reply mathesar/templates/mathesar/login_base.html:149]: I don't have an approach figured out yet. I figure let's get everything out for now, and then only add things back into the service if absolutely needed later. Couldn't we have a custom front page served by the demo microservice? It could pull the needed assets from the "real service". Not sure, though.

File: Dockerfile
Comment (mathemancer) [reply Dockerfile:120]: Fair enough.

**Comment by mathemancer** [comment]
@pavish I made the requested change.

**Review by Anish9901 — APPROVED** [review]
Looks good to me!

**Comment by Anish9901** [comment]
Waiting on your re-review @pavish, please merge this if you don't have any more concerns.

File: mathesar/templates/mathesar/login_base.html
Comment (pavish) [reply mathesar/templates/mathesar/login_base.html:149]: I think having a custom front page should be fine.

However, we currently show a banner on the app while someone is viewing the demo. This PR removes that banner as well. I'm cool with removing it and adding it later based on requirements. I do think this needs to be tracked in a separate issue.

**Review by pavish — APPROVED** [review]
@mathemancer Thanks, looks great!

We will have to create an issue to track this: https://github.com/mathesar-foundation/mathesar/pull/3538#discussion_r1568677772

**Pull request merged**
